piet-coregraphics/src/text.rs

```diff
@@ -149,6 +154,11 @@ impl CoreGraphicsTextLayoutBuilder {
         self.has_set_default_attrs = true;
         let whole_range = self.attr_string.range();
         let font = self.current_font();
+        let ascent = (font.ascent() + 0.5).floor();
```

Comment:
Any particular reason these aren't just `.round()`?

Also... I'm more than a little worried about this. Are they rounding to px values ("points" in Apple-speak) or device pixels? If the former, that's too coarse when dpi scaling is more than 1.

Reply:
I am following a convention for how apple calculates line height, based on this:
https://stackoverflow.com/questions/5511830/how-does-line-spacing-work-in-core-text-and-why-is-it-different-from-nslayoutm

Reply:
What a rabbithole indeed. I think that convention is wrong, but we're right to follow it :)